### PR TITLE
UCT/SELF: Expose UCT_MD_FLAG_NEED_RKEY.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

File: test/gtest/uct/uct_test.cc [hunk 1]
```diff
@@ -344,15 +344,21 @@ void uct_test::entity::mem_alloc(size_t length, uct_allocated_memory_t *mem,
                                      alloc_name, mem);
         ASSERT_UCS_OK(status);
 
-        rkey_buffer = malloc(md_attr().rkey_packed_size);
+        if (md_attr().cap.flags & UCT_MD_FLAG_NEED_RKEY) {
+            rkey_buffer = malloc(md_attr().rkey_packed_size);
```
Comment (hoopoepg) [inline test/gtest/uct/uct_test.cc]: what about check for non-NULL ptr?

File: src/uct/sm/self/self_md.c [hunk 2]
```diff
@@ -9,7 +9,8 @@
 static ucs_status_t uct_self_md_query(uct_md_h md, uct_md_attr_t *attr)
 {
     /* Dummy memory registration provided. No real memory handling exists */
-    attr->cap.flags         = UCT_MD_FLAG_REG;
+    attr->cap.flags         = UCT_MD_FLAG_REG |
```
Comment (hoopoepg) [inline src/uct/sm/self/self_md.c]: will it work with zero packed size?

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5462/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:mlx:retest

File: test/gtest/uct/uct_test.cc (hunk 1 above)
Comment (yosefe) [reply test/gtest/uct/uct_test.cc]: this was in prev code too, will fix

File: src/uct/sm/self/self_md.c (hunk 2 above)
Comment (yosefe) [reply src/uct/sm/self/self_md.c]: yes, even though rkey pack should not be called on this md in the first place

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5466/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@hoopoepg fixed

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5476/ for details (Mellanox internal link).

**Pull request merged**